ert3/storage/_storage.py

```diff
@@ -35,11 +35,16 @@ def init(*, workspace: Path) -> None:
             workspace=workspace,
             experiment_name=f"{workspace}.{special_key}",
             parameters=[],
+            ensemble_size=None,
         )
 
 
 def init_experiment(
-    *, workspace: Path, experiment_name: str, parameters: Iterable[str]
+    *,
+    workspace: Path,
+    experiment_name: str,
+    parameters: Iterable[str],
+    ensemble_size: Optional[int],
```

Comment:
Since all external calls to `init_experiment` should know the `ensemble_size` I suggest we drop `Optional` here. And for the call in `init` we pass `-1` explicitly.